ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+    activity.
+    (The graphs of activities for the child graph are created at the same time as the rest of the activities, but 
+    they are reinitialised and re-executed for each input row processed by the parent activity.)
+
+Other helper functions
+    The generated code contains other functions which are used to describe the meta information for the rows
+    processed within the graph.
+    E.g., the following class describes the output from the index read activity::
+    
+        struct mi1 : public CFixedOutputMetaData {
+            inline mi1() : CFixedOutputMetaData(120) {}
+            virtual const RtlTypeInfo * queryTypeInfo() const { return &ty1; }
+        } mx1;
+        
+    This represents a fixed size row which occupies 120 bytes.  The object returned by the queryTypeInfo() function
+    provides information about the types of the fields::
+    
+        const RtlStringTypeInfo ty2(0x4,40);
+        const RtlFieldStrInfo rf1("name",NULL,&ty2);
+        const RtlStringTypeInfo ty3(0x4,80);
+        const RtlFieldStrInfo rf2("address",NULL,&ty3);
+        const RtlFieldInfo * const tl4[] = { &rf1,&rf2, 0 };
+        const RtlRecordTypeInfo ty1(0xd,120,tl4);
+
+    I.e. a string column, length 40 called "name", followed by a string column, length 80 called "address".  The
+    interface IOutputMetaData in eclhelper.hpp is key to understanding how the rows are processed.
+
+Inline dataset operations
+    The rule mentioned at the start - that the generated code doesn't contain any knowledge of how to perform a
+    particular dataset operation - does have one notable exception.  Some operations on child datasets are very
+    simple to implement, and more efficient if they are implemented using inline c++ code.  (The generated code
+    is smaller, and they avoid the overhead setting up a child graph).  Examples include filtering and aggregating
+    column values from a child dataset.
+    
+The full code in the different engines is more complicated than the simplified process outlined above, especially
+when it comes to executing dependencies, but the broad outline is the same.
+
+Appendix
+========
+
+More information on the work done in the code generator to create the workunit can be found in ecl/eclcc/DOCUMENTATION.rst.
+
+The c++ code can be generated as a single c++ file or multiple files.  The system defaults to multiple c++ files, so that
+they can be compiled in parallel (and to avoid problems some compilers have with very large files).  When multipe c++ files are
+generated the metadata classes and workflow classes are generated in the main module, and the activities are generated
+in the files suffixed with a number.  It may be easier to understand the generated code if it is in one place.  In which case
+compile your query with the option -fspanMultipleCpp-.  Use -fsaveCppTempFiles to ensure the C++ files are not deleted
+(they will appear as helpers in the workunit details).
```

Comment:
'they' refers to? The options themselves (used here) or something relating to the C++ files?